samples/MvcSample.Web/HomeController.cs

```diff
@@ -90,6 +91,23 @@ public ActionResult SaveUser(User user)
         }
 
         /// <summary>
+        /// Action that shows multiple file upload.
+        /// </summary>
+        public async Task<ActionResult> PostFile(IList<IFormFile> file)
+        {
+            if (!ModelState.IsValid)
+            {
+                return View("MyView");
+            }
+
+            foreach (var f in file)
+            {
+                await f.SaveAsAsync("C:\\Temp\\test-file" + file.IndexOf(f));
```

Comment:
Let's change this code before it gets checked in. Have this Controller `Activate` an `IHostingEnvironment` instance and save the files to `Path.Combine(hostingEnvironment.WebRoot, "UploadedFiles", f.Name);`

cc @Tratcher should `SaveAsAsync` have an overload for IFileSystem `SaveAsAsync(IFileSystem, string path);` so you could do `SaveAsAsync(hostingEnvironment, Path.Combine("UploadFiles", f.Name));`